content/en/real_user_monitoring/session_replay/mobile/_index.md

```diff
@@ -18,26 +18,97 @@ Mobile Session Replay expands visibility into your mobile applications by visual
+
+{{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-1.png" alt="An example of how the iOS app screen contains 78 native views, but is made up of 19 wireframes." style="width:30%;">}}
+
+These wireframes can be sent to the player while **preserving their rendering order** (back-to-front) and **using absolute positioning** (in screen coordinates). There is no "tree structure", nor child-parent relationship between wireframes, therefore making the structure "flat".
+
+### Rendering algorithm
+
+To display a single replay frame, the player needs to iterate through all wireframes and render one another into the viewport using geometry information (x, y, width, height) from each wireframe. It must respect wireframes rendering order, so that succeeding ones overdraw existing portions of the viewport.
+
+The above screen can be reconstructed in 19 passes of the renderer:
+
+| Iteration | 1 | 2 | 3 |
+|-----------|---|---|---|
+| Viewport | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-1.png" alt="An example of a 'geometry' wireframe." style="width:100%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-2.png" alt="An example of an 'image' wireframe." style="width:100%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-3.png" alt="An example of a 'text' wireframe." style="width:100%;">}} |
+| Description | Draw "geometry" wireframe into viewport (rectangle + background color). | Draw "image" wireframe (rectangle + image with given UUID fetched from the Datadog backend). | Draw "text" wireframe (rectangle + "Twin Lake" text + font color, family and size). |
+
+The first wireframe could dictate the viewport size, enabling player size adjustment for given device + rotation (landscape / portrait).
+
+| Iteration | 4 | 5-9 | 10-12 |
+|-----------|---|---|---|
+| Viewport | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-4.png" alt="An example of a 'geometry', 'image, and 'text' wireframe." style="width:100%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-5-9-1.png" alt="An example of a 'geometry' and 'image' wireframe." style="width:100%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-10-12.png" alt="An example of a 'geometry' and 'image' wireframe." style="width:100%;">}} |
+| Description | Draw "image" and "text" wireframes into viewport.| Draw "image" and "text" wireframes into viewport. | Draw "image" and two "geometry" wireframes into viewport.  |
+
+Because wireframes are sorted in back-to-front order, it overdraws existing portion of the frame. This is very much desirable. For semi-transparent elements, it can also do the job of proper blending "closer" elements with the "farer" ones.
```

Comment:
```suggestion
Because wireframes are sorted in back-to-front order, it overdraws existing portion of the frame, which is a desirable behavior as it help support several UI patterns (for instance semi-transparent elements).
```